src/lib/work_partitioner/pending_zkapp_command.ml

```diff
@@ -0,0 +1,77 @@
+(* NOTE: this module is where the real optimization happen. One assumption we
+   have is the order of merging is irrelvant to the correctness of the final
+   proof. Hence we're only using a counter `merge_remaining` to track have we
+   reach the final proof
+*)
+
+open Core_kernel
+open Snark_work_lib
+
+type t =
+  { job : (Spec.Single.t, Id.Single.t) With_job_meta.t
+        (** the original work being splitted, should be identical to
```

Comment:
```suggestion
        (** the original work being split, should be identical to
```

Reply:
And it's not identical, just contains data from `Work_selector.work` and some metadata on top of it (most notably, "id").